src/federatedRemotesProvider.ts

```diff
@@ -90,10 +90,48 @@ export class FederatedRemotesProvider
       return Promise.resolve([]);
     }
 
-    const remoteEntrySyntaxTree = parseScript(remoteEntryBody);
+    const remoteEntryBodyLines = remoteEntryBody.split("\n");
+    const isDevMode = Boolean(
+      remoteEntryBodyLines.find((line) => line.includes('mode: "development"'))
+    );
+    const exposedEntries = this.extractExposedModules(remoteEntryBody, {
+      isDevMode,
+      remoteEntryBodyLines,
+    });
+
+    return exposedEntries.length === 0
+      ? Promise.resolve([])
+      : Promise.resolve(
+          exposedEntries.map((module: any) => {
+            const [modulePublicName, moduleLocalPath] = module;
+            return new FederatedRemoteTreeItem({
+              label: modulePublicName,
+              remoteEntry: moduleLocalPath,
+              collapsibleState: TreeItemCollapsibleState.None,
+              isExposedModule: true,
+            });
+          })
+        );
+  }
+
+  private getExposedModulesPropsInProdMode(remoteEntrySyntaxTree: any) {
+    let moduleMapDeclarator = null;
+
+    traverse(remoteEntrySyntaxTree, (node: any) => {
+      if (
+        node.type === Syntax.VariableDeclaration &&
+        node.declarations[0].id.name === "moduleMap"
+      ) {
+        moduleMapDeclarator = node.declarations[0];
+      }
+    });
 
+    return (moduleMapDeclarator as any).init.properties;
+  }
+
+  private getExposedModulesPropsInDevMode(remoteEntrySyntaxTree: any) {
     let expressionContainedModuleMap: any;
-    let moduleCode: string;
+    let moduleMapBody = "";
```

Comment:
not to nit pick but this (and all others) should be single quotes

Reply:
@zackarychapple okay, I've created a separate branch with new eslint rule and fixes all over the project in order to not mix with current changes